app/lib/shared/markdown.dart

```diff
@@ -219,18 +219,35 @@ class _RelativeUrlRewriter implements m.NodeVisitor {
 
   @override
   void visitElementAfter(m.Element element) {
+    // check current element
     if (element.tag == 'a') {
-      element.attributes['href'] = _rewriteUrl(element.attributes['href']);
+      _updateAttributes(element, 'href');
     } else if (element.tag == 'img') {
-      element.attributes['src'] =
-          _rewriteUrl(element.attributes['src'], raw: true);
+      _updateAttributes(element, 'src', raw: true);
+    }
+    // remove children that are marked to be removed
+    element.children?.removeWhere((n) =>
+        n is m.Element && n.attributes.containsKey('pub-markdown-remove'));
+  }
+
+  void _updateAttributes(m.Element element, String attrName,
+      {bool raw = false}) {
+    final newUrl = _rewriteUrl(element.attributes[attrName], raw: raw);
+    if (newUrl != null) {
+      element.attributes[attrName] = newUrl;
+    } else {
+      element.attributes.remove(attrName);
+      element.attributes['pub-markdown-remove'] = 'true';
```

Comment:
Would it not be safer to create a `final _elementsToRemove = <m.Element>[]` list on the `_RelativeUrlRewriter` class.

That way we don't abuse attribute names... we just have to make sure never reuse an instance of `_RelativeUrlRewriter`, because we've made it stateful, but that's should be no problem..

Also, shouldn't we replace the element with it's contents... so that the link disappears, but the contents of the link does not?

This only happens when you have invalid markdown? or an invalid URL... right?
Basically, any link with a URL we can't parse will have this property.. so shouldn't we just remove the link and retain the contents?

Example:
```markdown
 * [My awesome thing](http://illegal@@thing)
```

Won't this strip away the text "My awesome thing"? When we probably just want to strip away the `<a>` tag, or perhaps just the `href` attribute.

Reply:
Goog point, I've updated the code and the tests to match this. PTAL